### PR TITLE
ACMS-3639: Refactored DRS tests to remove RoboIo dependency.

### PR DESCRIPTION
**Pull request by deepakmishra2**
**Motivation**

Fixes #ACMS-3639

**Proposed changes**
Removed RoboIo dependency and created a new trait to fix output() method.

**Alternatives considered**

**Testing steps**

### CODE REVIEW

File: src/Common/IO.php [hunk 1]
```diff
@@ -88,7 +298,7 @@ protected function formatQuestion($message): string {
    * @return string
    *   The chosen option.
    */
-  protected function askChoice(string $question, array $options, mixed $default = NULL): string {
+  protected function askChoice($question, $options, $default = NULL): string {
```
Comment (rajeshreeputra) [inline src/Common/IO.php:301]: why this existing type hint removed?

File: src/Common/IO.php [hunk 2]
```diff
@@ -102,16 +312,16 @@ protected function askChoice(string $question, array $options, mixed $default =
    * @return string
    *   The response.
    */
-  protected function askRequired(string $message): string {
+  protected function askRequired($message): string {
```
Comment (rajeshreeputra) [inline src/Common/IO.php:315]: Here as well.

**Review by apathak18 — APPROVED** [review]

File: src/Common/IO.php (hunk 2 above)
Comment (deepakmishra2) [reply src/Common/IO.php]: added it back

File: src/Common/IO.php (hunk 1 above)
Comment (deepakmishra2) [reply src/Common/IO.php]: added it back

File: tests/src/Traits/DrsIO.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+   * @return $this
+   *
+   * @see \Robo\Contract\OutputAwareInterface::getOutput()
+   */
+  public function getOutput(): OutputInterface {
+    if (!isset($this->output)) {
+      $this->setOutput(new NullOutput());
+    }
+    return $this->output;
+  }
```
Comment (rajeshreeputra) [inline tests/src/Traits/DrsIO.php:219]: Everything appears to have been copied from Robo\IO. I am not convinced this is an optimal implementation.
cc @vishalkhode1

Comment (deepakmishra2) [reply tests/src/Traits/DrsIO.php:219]: @rajeshreeputra It sounds like you're discussing implementing a feature that's similar to something from Robo IO but using our own IO system for testing purposes. That sounds like a practical approach. By implementing it ourself, we have more control over the implementation and can tailor it to fit our specific needs, which could include better integration with our existing systems or customisations to suit our unique requirements. Testing it with our own IO system ensures that it behaves as expected within our environment. Overall, it sounds like a thoughtful approach to building and testing our system

@vishalkhode1 - i think we discussed on this approach as to use required methods and create our own IO as done in this approach

Comment (vishalkhode1) [reply tests/src/Traits/DrsIO.php:219]: Agree with @rajeshreeputra. The idea was here to not copy the complete code from Robo IO, but only the Required methods which will help us in Unit testing of class IO (provided by DRS).

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**